concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limiter/BlockingLimiter.java

```diff
@@ -17,34 +17,47 @@
 
 import com.netflix.concurrency.limits.Limiter;
 
+import java.time.Duration;
+import java.time.Instant;
 import java.util.Optional;
 
 /**
  * {@link Limiter} that blocks the caller when the limit has been reached.  The caller is
- * blocked until the limiter has been released.  This limiter is commonly used in batch
- * clients that use the limiter as a back-pressure mechanism.
+ * blocked until the limiter has been released, or a timeout is reached.  This limiter is
+ * commonly used in batch clients that use the limiter as a back-pressure mechanism.
  * 
  * @param <ContextT>
  */
 public final class BlockingLimiter<ContextT> implements Limiter<ContextT> {
     public static <ContextT> BlockingLimiter<ContextT> wrap(Limiter<ContextT> delegate) {
-        return new BlockingLimiter<>(delegate);
+        return new BlockingLimiter<>(delegate, Optional.empty());
+    }
+
+    public static <ContextT> BlockingLimiter<ContextT> wrap(Limiter<ContextT> delegate, Duration timeout) {
+        return new BlockingLimiter<>(delegate, Optional.of(timeout));
     }
 
     private final Limiter<ContextT> delegate;
+    private final Optional<Duration> timeout;
     
     /**
      * Lock used to block and unblock callers as the limit is reached
      */
     private final Object lock = new Object();
 
-    private BlockingLimiter(Limiter<ContextT> limiter) {
+    private BlockingLimiter(Limiter<ContextT> limiter, Optional<Duration> timeout) {
         this.delegate = limiter;
+        this.timeout = timeout;
     }
     
     private Optional<Listener> tryAcquire(ContextT context) {
+        Optional<Instant> timeoutInstant = timeout.map(t -> Instant.now().plus(t));
```

Comment:
Let's call this `deadline` instead instead of `timeoutInstant`.  Also, we can simplify the code by not using optional here.  So do something like `Instant deadline = timeout.map(t -> Instant.now().plus(t).orElse(Instant.MAX))`

Reply:
Deadline is great, I spent a while thinking about a good name for this but didn't consider, actioned.